core/src/cli_commands/vote_update_pool.rs

```diff
@@ -0,0 +1,436 @@
+        let new_pool_box_address_hash = force_any_val::<Digest32>();
+        let wallet = Wallet::from_secrets(vec![secret.clone().into()]);
+        let network_prefix = ergo_lib::ergotree_ir::chain::address::NetworkPrefix::Mainnet;
+        let change_address = AddressEncoder::new(network_prefix)
+            .parse_address_from_str("9iHyKxXs2ZNLMp9N9gbUT9V8gTbsV7HED1C1VhttMfBUMPDyF7r")
+            .unwrap();
+
+        let ballot_token_id = force_any_val::<TokenId>();
+        let ballot_token = Token {
+            token_id: ballot_token_id.clone(),
+            amount: 1.try_into().unwrap(),
+        };
+        let wallet_unspent_box = make_wallet_unspent_box(
+            secret.public_image(),
+            BoxValue::SAFE_USER_MIN
+                .checked_mul_u32(100_000_000)
+                .unwrap(),
+            Some(BoxTokens::from_vec(vec![ballot_token]).unwrap()),
+        );
+        let reward_token_id = force_any_val::<TokenId>();
+        let update_nft_token_id = force_any_val::<TokenId>();
+        let wallet_mock = WalletDataMock {
+            unspent_boxes: vec![wallet_unspent_box],
+        };
+        let unsigned_tx = build_tx_for_first_ballot_box(
+            &wallet_mock,
+            new_pool_box_address_hash,
+            reward_token_id,
+            100_000,
+            height - 3,
+            network_prefix,
+            10_000_000,
+            update_nft_token_id,
+            ballot_token_id,
+            "9iHyKxXs2ZNLMp9N9gbUT9V8gTbsV7HED1C1VhttMfBUMPDyF7r".into(),
+            height,
+            change_address,
+        )
+        .unwrap();
+
+        let tx_context = TransactionContext::new(
+            unsigned_tx.clone(),
+            find_input_boxes(unsigned_tx, wallet_mock.get_unspent_wallet_boxes().unwrap()),
+            Vec::new(),
+        )
+        .unwrap();
+
+        let _signed_tx = wallet.sign_transaction(tx_context, &ctx, None).unwrap();
+    }
+
+    #[test]
+    fn test_vote_update_pool_with_existing_ballot_box() {
+        let ctx = force_any_val::<ErgoStateContext>();
+        let height = ctx.pre_header.height;
+
+        let secret = force_any_val::<DlogProverInput>();
+        let new_pool_box_address_hash = force_any_val::<Digest32>();
+        let wallet = Wallet::from_secrets(vec![secret.clone().into()]);
+        let network_prefix = ergo_lib::ergotree_ir::chain::address::NetworkPrefix::Mainnet;
+        let change_address = AddressEncoder::new(network_prefix)
+            .parse_address_from_str("9iHyKxXs2ZNLMp9N9gbUT9V8gTbsV7HED1C1VhttMfBUMPDyF7r")
+            .unwrap();
+
+        let reward_token_id = force_any_val::<TokenId>();
+        let ballot_token_id = force_any_val::<TokenId>();
+        let ballot_token = Token {
+            token_id: ballot_token_id.clone(),
+            amount: 1.try_into().unwrap(),
+        };
+        let tokens = BoxTokens::from_vec(vec![ballot_token]).unwrap();
+        let in_ballot_box = ErgoBox::new(
```

Comment:
You can call `make_local_ballot_box_candidate` and then `ErgoBox::from_candidate(...)` afterwards.

Reply:
Done, thanks.